subprojects/version-control/src/main/java/org/gradle/vcs/VersionControlSystem.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.vcs;
+
+import java.io.File;
+
+/**
+ * Allows the user to perform generic version control operations in ways
+ * specified by the underlying implementations.
+ */
+public interface VersionControlSystem {
+    /**
+     * Populates the {@code workingDir} with the latest state of the
+     * version control repostory from the {@code spec}.
+     */
+    public void populate(File workingDir, VersionControlSpec spec);
```

Comment:
As an initial cut, I think this is probably fine to get some plumbing done.

We should think about the other operations we'll need...

- populate/initialize a copy of the repository
- update the repository
- checkout particular branches/commits/tags
- reset to a known state 
- list tags/branches?

Reply:
Thanks for the list. But, I really think we should do any of those things in a separate commit.